proto/rill/runtime/v1/resources.proto

```diff
@@ -296,13 +300,34 @@ message AlertSpec {
     string query_for_user_email = 12;
     google.protobuf.Struct query_for_attributes = 13;
   }
-  repeated string email_recipients = 14;
-  bool email_on_recover = 15;
-  bool email_on_fail = 16;
-  bool email_on_error = 17;
-  bool email_renotify = 18;
-  uint32 email_renotify_after_seconds = 19;
-  map<string, string> annotations = 20;
+  AlertNotifySpec notify_spec = 14;
+  map<string, string> annotations = 15;
+}
+
+message AlertNotifySpec {
+  bool notify_on_recover = 1;
+  bool notify_on_fail = 2;
+  bool notify_on_error = 3;
+  bool renotify = 4;
+  uint32 renotify_after_seconds = 5;
+  repeated NotifierSpec notifiers = 6;
+}
+
+message NotifierSpec {
+  oneof spec {
+    EmailNotifierSpec email = 1;
+    SlackNotifierSpec slack = 2;
+  }
+}
+
+message EmailNotifierSpec {
+  repeated string recipients = 1;
+}
+
+message SlackNotifierSpec {
+  repeated string emails = 1;
+  repeated string channels = 2;
+  repeated string webhooks = 3;
 }
```

Comment:
A few thoughts:
1. I would avoid calling this `NotifierSpec` because it sounds like a spec for a "notifier" resource, but no such resource exists. All the other `XXXSpec` types refer to resource types. Instead, I would propose calling it just `message Notifier` (similar to `message Schedule` which is referenced by multiple resource specs).
2. I feel ambivalent about hard-coding the properties for the different notification implementations (email, Slack, etc.). We did that with source properties and also analytical APIs, and we ended up refactoring both. It doesn't fit nicely with the extensible nature of our connectors (especially if we need to make some of them proprietary, i.e. stored in a private repo). I would consider mirroring what we did with source properties and with resolvers (e.g. see the `SourceSpec` and `APISpec` types), e.g.:
```proto
message Notifier {
  string connector = 1;
  google.protobuf.struct properties = 2;
}
```
3. Lastly, we now have a level of nesting where both the notifiers are repeated and the individual properties (like `emails` or `channels`) are repeated. Would it maybe make sense to only have `repeated` at the root to simplify things, and still have the same behavior? I'm thinking of this kind of refactor (not sure it's a good idea, just a suggestion):
```yaml
notify:
- connector: slack
   channel: #channel-foo
- connector: slack
   channel: #channel-bar
- connector: email
   recipient: a@example.com
- connector: email
   recipient: b@example.com
```

Reply:
As for the item 3, if a connector requires additional props passed, then these props are duplicated for each connector recipient (user/channel/webhook)